The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7783588829/job/21222350803) [(plain)](https://github.com/rust-lang/rust/commit/1d5e635d6a9bbf594f6ca3e316b884cf8556436a/checks/21222350803/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#12 writing image sha256:de9be95c84b310a9be14f4c803c4e14ae1219eb96b904e2e2013d3b8aa8cbfe9 done
#12 naming to docker.io/library/rust-ci done
#12 DONE 9.9s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Mon Feb  5 11:19:19 UTC 2024
  network time: Mon, 05 Feb 2024 11:19:19 GMT
  network time: Mon, 05 Feb 2024 11:19:19 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'build.optimized-compiler-builtins', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
i..i...................................i........................................ii...... 10120/16075
........................................................................................ 10208/16075
........................................................................................ 10296/16075
....................................................................................i... 10384/16075
..........................................i......................F...................... 10472/16075
...........................................i...F........................................ 10560/16075
.......................................................................................i 10736/16075
........................................................................................ 10824/16075
........................................................................................ 10912/16075
........................................i............................................... 11000/16075
---

---- [ui] tests/ui/parallel-rustc/cache-after-waiting-issue-111528.rs stdout ----
diff of stderr:

1 error: symbol `fail` is already defined
+   --> $DIR/cache-after-waiting-issue-111528.rs:12:1
3    |
4 LL | pub fn b() {
5    | ^^^^^^^^^^
---
To only update this specific test, also pass `--test-args parallel-rustc/cache-after-waiting-issue-111528.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parallel-rustc/cache-after-waiting-issue-111528" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parallel-rustc/cache-after-waiting-issue-111528/auxiliary" "-Z" "threads=16"
--- stderr -------------------------------
--- stderr -------------------------------
error: symbol `fail` is already defined
##[error]  --> /checkout/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.rs:12:1
LL | pub fn b() {
   | ^^^^^^^^^^

error: aborting due to 1 previous error
error: aborting due to 1 previous error
------------------------------------------
Build completed unsuccessfully in 0:12:54


---- [ui] tests/ui/parallel-rustc/workerlocal-outer-thread-pool-issue-94654.rs stdout ----

error: Error: expected failure status (Some(1)) but received status None.
status: signal: 6 (SIGABRT) (core dumped)
command: RUSTC_ICE="0" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/parallel-rustc/workerlocal-outer-thread-pool-issue-94654.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "--sysroot" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2" "--target=x86_64-unknown-linux-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parallel-rustc/workerlocal-outer-thread-pool-issue-94654" "-A" "unused" "-A" "internal_features" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "-L" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/parallel-rustc/workerlocal-outer-thread-pool-issue-94654/auxiliary" "-Z" "threads=16"
--- stderr -------------------------------
--- stderr -------------------------------
error[E0072]: recursive type `ElemDerived` has infinite size
##[error]  --> /checkout/tests/ui/parallel-rustc/workerlocal-outer-thread-pool-issue-94654.rs:6:1
LL | pub enum ElemDerived {
   | ^^^^^^^^^^^^^^^^^^^^
...
...
LL |     A(ElemDerived)
   |       ----------- recursive without indirection
   |
help: insert some indirection (e.g., a `Box`, `Rc`, or `&`) to break the cycle
   |
LL |     A(Box<ElemDerived>)

thread 'rustc' panicked at /checkout/compiler/rustc_middle/src/ty/context/tls.rs:111:50:
no ImplicitCtxt stored in tls
stack backtrace:
stack backtrace:
   0:     0x7f14154680e9 - <std[c4bfee2854751784]::sys_common::backtrace::_print::DisplayBacktrace as core[bfa390d9d4a9e81f]::fmt::Display>::fmt
   1:     0x7f14154c20bf - core[bfa390d9d4a9e81f]::fmt::write
   2:     0x7f141545c955 - <std[c4bfee2854751784]::sys::pal::unix::stdio::Stderr as std[c4bfee2854751784]::io::Write>::write_fmt
   3:     0x7f1415467ec5 - std[c4bfee2854751784]::sys_common::backtrace::print
   4:     0x7f141546b21a - std[c4bfee2854751784]::panicking::default_hook::{closure#1}
   5:     0x7f141546af1d - std[c4bfee2854751784]::panicking::default_hook
   6:     0x7f1416102427 - <alloc[5c38b0e5f86a9b1e]::boxed::Box<rustc_driver_impl[f91a60e74c7bb87d]::install_ice_hook::{closure#0}> as core[bfa390d9d4a9e81f]::ops::function::Fn<(&dyn for<'a, 'b> core[bfa390d9d4a9e81f]::ops::function::Fn<(&'a core[bfa390d9d4a9e81f]::panic::panic_info::PanicInfo<'b>,), Output = ()> + core[bfa390d9d4a9e81f]::marker::Sync + core[bfa390d9d4a9e81f]::marker::Send, &core[bfa390d9d4a9e81f]::panic::panic_info::PanicInfo)>>::call
   7:     0x7f141546b8d4 - std[c4bfee2854751784]::panicking::rust_panic_with_hook
   8:     0x7f141546b662 - std[c4bfee2854751784]::panicking::begin_panic_handler::{closure#0}
   9:     0x7f1415468696 - std[c4bfee2854751784]::sys_common::backtrace::__rust_end_short_backtrace::<std[c4bfee2854751784]::panicking::begin_panic_handler::{closure#0}, !>
  11:     0x7f1415421ca5 - core[bfa390d9d4a9e81f]::panicking::panic_fmt
  12:     0x7f1415421c63 - core[bfa390d9d4a9e81f]::option::expect_failed
  13:     0x7f141611beb6 - rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals::<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#2}
  13:     0x7f141611beb6 - rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals::<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#2}
  14:     0x7f14160b4aab - <rayon_core[6e4a57e9cc502a]::sleep::Sleep>::sleep
  15:     0x7f14160b3851 - <rayon_core[6e4a57e9cc502a]::registry::WorkerThread>::wait_until_cold
  16:     0x7f14194bdfad - <rayon_core[6e4a57e9cc502a]::registry::ThreadBuilder>::run
  17:     0x7f14160f6a17 - <scoped_tls[a944d82e16a82687]::ScopedKey<rustc_span[85514a83ff106e10]::SessionGlobals>>::set::<rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#0}::{closure#0}, ()>
  18:     0x7f1416168e0c - rustc_span[85514a83ff106e10]::set_session_globals_then::<(), rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#0}::{closure#0}>
  19:     0x7f14161003ec - <<crossbeam_utils[67620c07dccad3b6]::thread::ScopedThreadBuilder>::spawn<<rayon_core[6e4a57e9cc502a]::ThreadPoolBuilder>::build_scoped<rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#0}, rustc_interface[495028258ec9b88d]::util::run_in_thread_pool_with_globals<rustc_interface[495028258ec9b88d]::interface::run_compiler<core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>, rustc_driver_impl[f91a60e74c7bb87d]::run_compiler::{closure#0}>::{closure#0}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#3}::{closure#0}::{closure#1}, core[bfa390d9d4a9e81f]::result::Result<(), rustc_span[85514a83ff106e10]::ErrorGuaranteed>>::{closure#0}::{closure#0}::{closure#0}, ()>::{closure#0} as core[bfa390d9d4a9e81f]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  20:     0x7f14161022d2 - <alloc[5c38b0e5f86a9b1e]::boxed::Box<dyn core[bfa390d9d4a9e81f]::ops::function::FnOnce<(), Output = ()> + core[bfa390d9d4a9e81f]::marker::Send> as core[bfa390d9d4a9e81f]::ops::function::FnOnce<()>>::call_once
  21:     0x7f141611b426 - std[c4bfee2854751784]::sys_common::backtrace::__rust_begin_short_backtrace::<alloc[5c38b0e5f86a9b1e]::boxed::Box<dyn core[bfa390d9d4a9e81f]::ops::function::FnOnce<(), Output = ()> + core[bfa390d9d4a9e81f]::marker::Send>, ()>
  22:     0x7f141616b599 - std[c4bfee2854751784]::panicking::try::<(), core[bfa390d9d4a9e81f]::panic::unwind_safe::AssertUnwindSafe<<std[c4bfee2854751784]::thread::Builder>::spawn_unchecked_<alloc[5c38b0e5f86a9b1e]::boxed::Box<dyn core[bfa390d9d4a9e81f]::ops::function::FnOnce<(), Output = ()> + core[bfa390d9d4a9e81f]::marker::Send>, ()>::{closure#1}::{closure#0}>>
  23:     0x7f141611c3dc - <<std[c4bfee2854751784]::thread::Builder>::spawn_unchecked_<alloc[5c38b0e5f86a9b1e]::boxed::Box<dyn core[bfa390d9d4a9e81f]::ops::function::FnOnce<(), Output = ()> + core[bfa390d9d4a9e81f]::marker::Send>, ()>::{closure#1} as core[bfa390d9d4a9e81f]::ops::function::FnOnce<()>>::call_once::{shim:vtable#0}
  24:     0x7f1415476b7e - <std[c4bfee2854751784]::sys::pal::unix::thread::Thread>::new::thread_start
  26:     0x7f14152ef120 - <unknown>
  27:                0x0 - <unknown>

error: the compiler unexpectedly panicked. this is a bug.
error: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/issues/new?labels=C-bug%2C+I-ICE%2C+T-compiler&template=ice.md

note: rustc 1.78.0-nightly (fab6941cf 2024-02-05) running on x86_64-unknown-linux-gnu

note: compiler flags: -Z threads=1 -Z simulate-remapped-rust-src-base=/rustc/FAKE_PREFIX -Z translate-remapped-path-to-local-path=no -Z ignore-directory-in-diagnostics-source-blocks=/cargo -C codegen-units=1 -Z ui-testing -Z deduplicate-diagnostics=no -Z write-long-types-to-disk=no -C strip=debuginfo -C prefer-dynamic -C rpath -C debuginfo=0 -Z threads=16
query stack during panic:
end of query stack
end of query stack
Rayon: detected unexpected panic; aborting



failures:
```

</details>

--- a/src/tools/tidy/src/ui_tests.rs
+++ b/src/tools/tidy/src/ui_tests.rs
@@ -15,7 +15,7 @@ use std::path::{Path, PathBuf};
 const ENTRY_LIMIT: usize = 900;
 // FIXME: The following limits should be reduced eventually.
 const ISSUES_ENTRY_LIMIT: usize = 1819;
-const ROOT_ENTRY_LIMIT: usize = 870;
+const ROOT_ENTRY_LIMIT: usize = 871;
 
 const EXPECTED_TEST_FILE_EXTENSIONS: &[&str] = &[
     "rs",     // test source files

--- a/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.rs
+++ b/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.rs
@@ -1,0 +1,16 @@
+// compile-flags: -Z threads=16
+// build-fail
+
+#![crate_type="rlib"]
+#![allow(warnings)]
+
+#[export_name="fail"]
+pub fn a() {
+}
+
+#[export_name="fail"]
+pub fn b() {
+//~^ Error symbol `fail` is already defined
+}
+
+fn main() {}

--- a/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.stderr
+++ b/tests/ui/parallel-rustc/cache-after-waiting-issue-111528.stderr
@@ -1,0 +1,8 @@
+error: symbol `fail` is already defined
+  --> $DIR/cache-after-waiting-issue-111528.rs:12:1
+   |
+LL | pub fn b() {
+   | ^^^^^^^^^^
+
+error: aborting due to 1 previous error
+

--- a/tests/ui/parallel-rustc/export-symbols-deadlock-issue-118205-2.rs
+++ b/tests/ui/parallel-rustc/export-symbols-deadlock-issue-118205-2.rs
@@ -1,0 +1,7 @@
+// compile-flags:-C extra-filename=-1 -Z threads=16
+// no-prefer-dynamic
+// build-pass
+#![crate_name = "crateresolve1"]
+#![crate_type = "lib"]
+
+pub fn f() -> isize { 10 }

--- a/tests/ui/parallel-rustc/export-symbols-deadlock-issue-118205.rs
+++ b/tests/ui/parallel-rustc/export-symbols-deadlock-issue-118205.rs
@@ -1,0 +1,22 @@
+// compile-flags: -Z threads=16
+// build-pass
+
+pub static GLOBAL: isize = 3;
+
+static GLOBAL0: isize = 4;
+
+pub static GLOBAL2: &'static isize = &GLOBAL0;
+
+pub fn verify_same(a: &'static isize) {
+    let a = a as *const isize as usize;
+    let b = &GLOBAL as *const isize as usize;
+    assert_eq!(a, b);
+}
+
+pub fn verify_same2(a: &'static isize) {
+    let a = a as *const isize as usize;
+    let b = GLOBAL2 as *const isize as usize;
+    assert_eq!(a, b);
+}
+
+fn main() {}

--- a/tests/ui/parallel-rustc/hello_world.rs
+++ b/tests/ui/parallel-rustc/hello_world.rs
@@ -1,0 +1,6 @@
+// compile-flags: -Z threads=8
+// run-pass
+
+fn main() {
+    println!("Hello world!");
+}

--- a/tests/ui/parallel-rustc/read-stolen-value-issue-111520.rs
+++ b/tests/ui/parallel-rustc/read-stolen-value-issue-111520.rs
@@ -1,0 +1,18 @@
+// compile-flags: -Z threads=16
+// run-pass
+
+#[repr(transparent)]
+struct Sched {
+    i: i32,
+}
+impl Sched {
+    extern "C" fn get(self) -> i32 { self.i }
+}
+
+fn main() {
+    let s = Sched { i: 4 };
+    let f = || -> i32 {
+        s.get()
+    };
+    println!("f: {}", f());
+}

